sdk/src/main/java/io/radar/sdk/RadarJobScheduler.kt

```diff
@@ -40,21 +42,58 @@ class RadarJobScheduler : JobService() {
                 putString(EXTRA_SOURCE, source.name)
             }
 
-            val jobInfo = JobInfo.Builder(JOB_ID, componentName)
+            val settings = RadarSettings.getFeatureSettings(context)
+            val newMaxConcurrentJobs = settings.maxConcurrentJobs
+            val currentMax = maxConcurrentJobs
+            // This block ensures the number of "available" jobs remains limited to the size of this map.
+            when {
+                currentMax == null -> {
+                    // First assignment
+                    for (i in 0..newMaxConcurrentJobs) {
+                        counter[JOB_ID + i] = 0
+                    }
+                }
+                currentMax < newMaxConcurrentJobs -> {
+                    // More jobs available
+                    for (i in currentMax..newMaxConcurrentJobs) {
+                        counter[JOB_ID + i] = 0
+                    }
+                }
+                currentMax > newMaxConcurrentJobs -> {
+                    // Less jobs available
+                    for (i in currentMax..newMaxConcurrentJobs) {
+                        counter.remove(JOB_ID + i)
+                    }
+                }
+            }
+            maxConcurrentJobs = newMaxConcurrentJobs
+
+            val job = if (counter.any { it.value == 0 }) {
+                counter.minByOrNull { it.value }!!
+            } else {
+                // Overwrites the longest-scheduled-job
+                counter.maxByOrNull { it.value }!!
+            }
+
+            val jobId = job.key
+            val jobInfo = JobInfo.Builder(jobId, componentName)
                 .setExtras(extras)
                 .setMinimumLatency(0)
                 .setOverrideDeadline(0)
+                .setRequiredNetworkType(
+                    if (settings.schedulerRequiresNetwork) JobInfo.NETWORK_TYPE_ANY else JobInfo.NETWORK_TYPE_NONE
+                )
                 .build()
 
-            val jobScheduler = context.getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
+            val jobScheduler = context.getSystemService(JOB_SCHEDULER_SERVICE) as JobScheduler
             val result = jobScheduler.schedule(jobInfo)
             if (result == JobScheduler.RESULT_SUCCESS) {
                 Radar.logger.d(
                     "Scheduling Location Job |" +
                             " location = $location;" +
                             " source = ${stringForSource(source)};" +
                             " success = true;" +
-                            " overwrites = ${counter.getAndIncrement()}"
+                            " overwrites = ${counter[jobId]}"
```

Comment:
I think this might be missing an increment?  Or maybe I missed it somewhere else.

Reply:
Good catch. See update.